Adding deviation for isisLevelEnabled, max_ecmp_paths for Juniper routers

priyaranjannanda opened the pull request:
Changes include : 

1. Below deviations are required for juniper routers as currently it doesn’t support the ISIS without setting level enable and doesn’t support configuring max ecmp paths for ISIS as 2. Hence adding devitaion 

-     isis_level_enabled: true

-     max_ecmp_paths: true
2. Script Fix:
       i.      Changed load balance percent to 50/50 in the code.
      ii.      While calculating the balance, subtracted the traffic received for profile pfx1ToPfx4 in lag2 and calculated the weight using rest of the traffic in lag2.
     iii.      Changed the packet size to 1400.

ram-mac commented:
Hello,
 Please address following comments
 1) The static analysis is failing
 2) The traffic distribution weights are change to 50/50 was 33/67 earlier. What is the reason for change?
 3) The frame size is changed to 1400 from 1500; was wondering if the test fails with 1500 frame size?

ram-mac commented:
Hi,
 I fixed the ln1206 where it is failing "rxs = append(rxs, inFrames)" which originally is "rxs = append(rxs, inOctets)"
 However, after this fix also the test fails. Please let me know if it is passing at ur end.

priyaranjannanda commented:
> Hi, I fixed the ln1206 where it is failing "rxs = append(rxs, inFrames)" which originally is "rxs = append(rxs, inOctets)" However, after this fix also the test fails. Please let me know if it is passing at ur end.

Hi,

I have changed it to inFrames and it is passing now.

--
Priya Ranjan

priyaranjannanda commented:
> Hello, Please address following comments
> 
> 1. The static analysis is failing
> 2. The traffic distribution weights are change to 50/50 was 33/67 earlier. What is the reason for change?
> 3. The frame size is changed to 1400 from 1500; was wondering if the test fails with 1500 frame size?

Hi,

1. I have uploaded new changes and  static analysis has passed.

2 & 3 . 
There are 2 traffic profiles (pfx1ToPfx2_3, pfx1ToPfx4) transmitting traffic to lag2 and one profile (pfx1ToPfx2_3 ) transmitting traffic to lag3.

Load balance percent per readme is 50/50 , but testcase is coded with 33/67 for lag2/lag3 respectively.
When weight (load balance) is calculated for pfx1ToPfx2_3, entire traffic to lag2 and lag3 are considered.
Traffic is sent with fixed packet size of 1500 bytes and default interface mtu is 1514. gRIBI route is programmed to encap traffic as IPinIP which leads to fragmentation of packet in the egress lag2 interface.

While calculating the balance, subtracted the traffic received for profile pfx1ToPfx4 in lag2 and calculated the weight using rest of the traffic in lag2.


https://github.com/openconfig/featureprofiles/blob/main/feature/interface/aggregate/otg_tests/aggregate_all_not_viable_test/README.md#rt-5721-make-the-forwarding-viable-transitions-from-true----false-on-ports-3-6-within-the-lag_2-on-the-dut

ram-mac commented:
hi,
Can we fix the conflicts; I think some other changes has gone into aggregate_all_not_forwarding_viable_test.go test

priyaranjannanda commented:
> hi, Can we fix the conflicts; I think some other changes has gone into aggregate_all_not_forwarding_viable_test.go test

Hi Ram,

I have resolved the conflict. Please re-review.

--
Thanks & Regards,
Priya Ranjan

ram-mac commented:
> > hi, Can we fix the conflicts; I think some other changes has gone into aggregate_all_not_forwarding_viable_test.go test
> 
> Hi Ram,
> 
> I have resolved the conflict. Please re-review.
> 
> -- Thanks & Regards, Priya Ranjan

Hi Priya,
 The static analysis is failing. It looks to be some formatting issue. Please fix it
-Ram

priyaranjannanda commented:
> > > hi, Can we fix the conflicts; I think some other changes has gone into aggregate_all_not_forwarding_viable_test.go test
> > 
> > 
> > Hi Ram,
> > I have resolved the conflict. Please re-review.
> > -- Thanks & Regards, Priya Ranjan
> 
> Hi Priya, The static analysis is failing. It looks to be some formatting issue. Please fix it -Ram

Hi Ram,

I have fixed the formatting issue and static check has passed. Please review.

--
Thanks & Regards,
Priya Ranjan

ram-mac commented:
Looks Good now.

ram-mac reviewed (approved):
LGTM

self-maurya commented on feature/interface/aggregate/otg_tests/aggregate_all_not_viable_test/aggregate_all_not_forwarding_viable_test.go:
```diff
@@ -467,7 +467,8 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) []string {
 		agg3.ateLagCount = 2
-		if dut.Vendor() != ondatra.CISCO {
+		trafficDistributionWeights = []uint64{50, 50}
+		if dut.Vendor() != ondatra.CISCO && dut.Vendor() != ondatra.JUNIPER {
```
Vendor based should not be used unless wrapper under deviation.

Please use deviation based on this reference https://github.com/openconfig/featureprofiles/blob/main/internal/deviations/README.md#when-to-use-deviations
or if the feature is not supported then we can keep it failing.
